internal/internal.go

```diff
@@ -758,3 +758,21 @@ func v2Status(s *ext_type_v3.HttpStatus) *ext_type_v2.HttpStatus {
 		Code: ext_type_v2.StatusCode(s.Code),
 	}
 }
+
+// checkCtxError checks if the context has error and returns appropriate internal error
+func checkCtxError(ctx context.Context) (*Error, bool) {
```

Comment:
I believe this function can be simplified by just returning a `*Error` type and omitting the `bool` return value, since the `bool` is always true when the `*Error` is not nil, but false when the `*Error` is nil. Is there some other information you were hoping to use this `bool` to represent?